### PR TITLE
feat: Add country-code to metric tags for ad fill instrumentation

### PR DESCRIPTION
**Pull request by ncloudioj**
This fixes [DISCO-2293](https://mozilla-hub.atlassian.net/browse/DISCO-2293).

[DISCO-2293]: https://mozilla-hub.atlassian.net/browse/DISCO-2293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ncloudioj** [comment]
Hey @pjenvey, I was under the impression that we have some concerns about increasing metric cardinality when adding new tags to metrics, wanted to check that with you. This addition allows DS & AdOps to instrument ad fill rate (by country) of adM's tiles endpoint.

**Comment by pjenvey** [comment]
It looks like `add_metric` is intended for adding Sentry extra data (it should really be renamed), so you definitely want `add_tag`.

We only query adM for countries with an entry in the inclusion filter. That's currently a pretty small number which I don't think make this a high cardinality tag (and that number could probably at least double and still not matter).

Will this new tag be utilized temporarily (during investigation)? As the number of countries will likely grow -- you probably have a better sense than me of where that number might go. I'm not sure what the exact threshold for high cardinality is, it might depend on the service and how many metrics it emits. SRE probably has a better idea.

You can run these to resolve the cargo audits:
```shell
cargo update -p openssl
cargo update -p tempfile  # removes the remove_dir_all crate entirely
cargo update -p spin@0.9.4  # quiets the yanked warning
```

**Comment by ncloudioj** [comment]
> Will this new tag be utilized temporarily (during investigation)? As the number of countries will likely grow -- you probably have a better sense than me of where that number might go. I'm not sure what the exact threshold for high cardinality is, it might depend on the service and how many metrics it emits. SRE probably has a better idea.

Thanks for the context, Phil! This tag will be used permanently for us to monitor the ad fill rate since there is a contractual obligation on the partner.  Yup, looks like the increase should still be manageable but I will double check with SRE to be sure.

I updated the dependencies in #525, will rebase once that gets merged.

**Review by pjenvey — DISMISSED** [review]

File: src/web/test.rs [hunk 1]
```diff
@@ -880,6 +880,7 @@ async fn metrics() {
     assert!(get_metric.contains("ua.form_factor:desktop"));
     assert!(get_metric.contains("ua.os.family:windows"));
     assert!(!&metrics[1].contains("endpoint:mobile"));
+    assert!(&metrics[1].contains("geo.country_code"));
```
[the remark this answers is not in the document]
Comment (ncloudioj) [reply src/web/test.rs:883]: Yup, `filter.adm.request` and `filter.adm.empty_response` are of interest here. So I added this tag to all metrics in `get_tiles()` and the subsequent ones. :)

**Comment by ncloudioj** [comment]
Thank you both for review!

**Pull request merged**